fuzz/fuzz_targets/cranelift-fuzzgen.rs

```diff
@@ -56,7 +59,13 @@ fuzz_target!(|testcase: TestCase| {
         let mut env = FunctionStore::default();
         env.add(testcase.func.name.to_string(), &testcase.func);
 
-        let state = InterpreterState::default().with_function_store(env);
+        let state = InterpreterState::default()
+            .with_function_store(env)
+            .with_libcall(LibCall::IshlI64, &|args| match &args[..] {
+                [DataValue::I64(_), DataValue::I64(b)] if *b > 63 => smallvec![DataValue::I64(0)],
+                [DataValue::I64(a), DataValue::I64(b)] => smallvec![DataValue::I64(a.shl(b))],
+                _ => unreachable!(),
+            });
```

Comment:
I saw some sources online that point to this being the correct behaviour, but nothing definitive. Does anyone know if this is correct?

Reply:
I think this is undefined behaviour, so I've added trap codes to the libcall handlers and reverted back to div

Reply:
I'm not sure I understand. We've recently defined shifts to be computed modulo the bit width. So I think `Ishl64` is a fine choice, but it should evaluate to `a << (b % 64)` for all values of `b`. That in turn means that we don't need to check the preconditions on Rust's implementation of left-shift: it would panic in debug builds if the shift amount is 64 or higher, but we'll never hit that case.

Reply:
These libcalls end up being linked to the platform's runtime library if the interpreted version succeeds, so we need to make sure that they match the behaviour agreed by the runtime libraries and not our own.

We also can't really panic here since it would cause a fuzz error.

Reply:
Another option would be to define our own symbol for the JIT as well, and then we control both sides, the interpreted version and the compiled version.